tests/lexer/tstrnumlits.nim

```diff
@@ -0,0 +1,151 @@
+# Test tkStrNumLit
+
+import std/macros
+import strutils
+import mlexerutils
+
+type
+  S = distinct string
+
+proc `'wrap`(number: string): S =
+  result = ("[[" & number & "]]").S
+
```

Comment:
```suggestion
proc `'wrap`(number: string): string =
  "{" & number & "}"
```
then the tests below simplify. Simplicity matters so we can easily add more tests, makes them more readable etc

(but maybe add just 1 single test with the previous version, say a wrap2, as sanity check)